pandas/tests/arrays/categorical/test_constructors.py

```diff
@@ -643,3 +645,55 @@ def test_constructor_string_and_tuples(self):
         c = pd.Categorical(np.array(["c", ("a", "b"), ("b", "a"), "c"], dtype=object))
         expected_index = pd.Index([("a", "b"), ("b", "a"), "c"])
         assert c.categories.equals(expected_index)
+
+    def test_from_dummies(self):
```

Comment:
Can you parametrize this over sparse = True/False?

Reply:
For the sake of simplicity I have not made an attempt to support loading from a sparse array but I can give it a go!